superset/security/manager.py

```diff
@@ -1337,13 +1341,22 @@ def get_guest_user_from_request(self, req: Request) -> Optional[GuestUser]:
             return None
 
         try:
+            aud = (
+                current_app.config["GUEST_TOKEN_JWT_DECODE_AUDIENCE"] or get_url_host()
+            )
             token = self.parse_jwt_guest_token(raw_token)
             if token.get("user") is None:
                 raise ValueError("Guest token does not contain a user claim")
             if token.get("resources") is None:
                 raise ValueError("Guest token does not contain a resources claim")
             if token.get("rls_rules") is None:
                 raise ValueError("Guest token does not contain an rls_rules claim")
+            if token.get("aud") is None:
+                raise ValueError("Guest token does not contain an aud claim")
+            if token.get("aud") != aud:
+                raise ValueError("Guest token does not match the aud claim")
+            if token.get("type") != "guest":
```

Comment:
Pls add some tests where the supplied token isn't a "guest" token, and where the `aud` doesn't match